building status-go fails in status-mobile due to missing generated files

## Summary

Since generated files were removed in this PR -> 

the nix derivation that builds status-go for mobile in status-mobile repo fails like this : 

```
siddarthkumar in ~/code/experiments/status-mobile on fix-status-go-generated-files λ make status-go-android                                                                                                        1m26s
nix/scripts/build.sh targets.status-go.mobile.android
trace: WARNING: Marking build version as 'develop'!
Running: nix-build --pure --fallback --no-out-link --show-trace --attr targets.status-go.mobile.android
trace: WARNING: Marking build version as 'develop'!
these 2 derivations will be built:
  /nix/store/1lk8yfv53shdbbyhrwb3lcrs1q0gsnps-status-go-230e05c-source.drv
  /nix/store/qzi5y7i1p2qqd7g0g0xz767fcxaxsd9y-status-go-develop-230e05c-android.drv
building '/nix/store/1lk8yfv53shdbbyhrwb3lcrs1q0gsnps-status-go-230e05c-source.drv'...

trying https://github.com/status-im/status-go/archive/230e05cd31e3f08605d5916434e4428671fa5b08.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 57.6M    0 57.6M    0     0  7708k      0 --:--:--  0:00:07 --:--:-- 6630k
unpacking source archive /private/tmp/nix-build-status-go-230e05c-source.drv-0/230e05cd31e3f08605d5916434e4428671fa5b08.tar.gz
building '/nix/store/qzi5y7i1p2qqd7g0g0xz767fcxaxsd9y-status-go-develop-230e05c-android.drv'...
Running phase: unpackPhase
unpacking source archive /nix/store/gdcsvl3iyr2nhy4bvydjjggmllvk4111-status-go-230e05c-source
source root is status-go-230e05c-source
Running phase: patchPhase
Running phase: configurePhase
Running phase: buildPhase

Building status-go for: android/arm,android/arm64,android/386
/nix/store/rzyfh6ibdc8q0s13g0z0sb2sxbnjzlh1-gomobile-unstable-2022-05-18/bin/gomobile: go build -tags gowaku_skip_migrations,gowaku_no_rln -ldflags -X github.com/status-im/status-go/params.GitCommit=230e05cd31e3f08605d5916434e4428671fa5b08 -X github.com/status-im/status-go/params.IpfsGatewayURL=https://ipfs.status.im/ -X github.com/status-im/status-go/params.Version=develop -s -w -buildid= -buildmode=c-shared -o=/private/tmp/nix-build-status-go-develop-230e05c-android.drv-0/gomobile-work/android/src/main/jniLibs/armeabi-v7a/libgojni.so ./gobind failed: exit status 1
../../go/src/github.com/status-im/status-go/mailserver/mailserver_db_postgres.go:18:2: cannot find package "github.com/status-im/status-go/mailserver/migrations" in any of:
        /private/tmp/nix-build-status-go-develop-230e05c-android.drv-0/go/src/github.com/status-im/status-go/vendor/github.com/status-im/status-go/mailserver/migrations (vendor tree)
        /nix/store/3l7m8vni45h2inv7fcifvp6izxm243an-go-1.21.10/share/go/src/github.com/status-im/status-go/mailserver/migrations (from $GOROOT)
        /private/tmp/nix-build-status-go-develop-230e05c-android.drv-0/gomobile-work/src/github.com/status-im/status-go/mailserver/migrations (from $GOPATH)
        /private/tmp/nix-build-status-go-develop-230e05c-android.drv-0/go/src/github.com/status-im/status-go/mailserver/migrations
        /nix/store/rzyfh6ibdc8q0s13g0z0sb2sxbnjzlh1-gomobile-unstable-2022-05-18/src/github.com/status-im/status-go/mailserver/migrations
package gobind
        imports github.com/status-im/status-go/mobile
        imports github.com/status-im/status-go/api
        imports github.com/status-im/status-go/node
        imports github.com/status-im/status-go/services/accounts
        imports github.com/status-im/status-go/protocol
        imports github.com/status-im/status-go/protocol/anonmetrics
        imports github.com/status-im/status-go/protocol/anonmetrics/migrations: no Go files in /private/tmp/nix-build-status-go-develop-230e05c-android.drv-0/go/src/github.com/status-im/status-go/protocol/anonmetrics/migrations
../../go/src/github.com/status-im/status-go/transactions/testhelpers.go:14:2: cannot find package "github.com/status-im/status-go/rpc/chain/mock/client" in any of:
        /private/tmp/nix-build-status-go-develop-230e05c-android.drv-0/go/src/github.com/status-im/status-go/vendor/github.com/status-im/status-go/rpc/chain/mock/client (vendor tree)
        /nix/store/3l7m8vni45h2inv7fcifvp6izxm243an-go-1.21.10/share/go/src/github.com/status-im/status-go/rpc/chain/mock/client (from $GOROOT)
        /private/tmp/nix-build-status-go-develop-230e05c-android.drv-0/gomobile-work/src/github.com/status-im/status-go/rpc/chain/mock/client (from $GOPATH)
        /private/tmp/nix-build-status-go-develop-230e05c-android.drv-0/go/src/github.com/status-im/status-go/rpc/chain/mock/client
        /nix/store/rzyfh6ibdc8q0s13g0z0sb2sxbnjzlh1-gomobile-unstable-2022-05-18/src/github.com/status-im/status-go/rpc/chain/mock/client
package gobind
        imports github.com/status-im/status-go/mobile
        imports github.com/status-im/status-go/api
        imports github.com/status-im/status-go/node
        imports github.com/status-im/status-go/services/accounts
        imports github.com/status-im/status-go/protocol
        imports github.com/status-im/status-go/protocol/sqlite
        imports github.com/status-im/status-go/protocol/encryption/migrations: no Go files in /private/tmp/nix-build-status-go-develop-230e05c-android.drv-0/go/src/github.com/status-im/status-go/protocol/encryption/migrations
package gobind
        imports github.com/status-im/status-go/mobile
        imports github.com/status-im/status-go/api
        imports github.com/status-im/status-go/node
        imports github.com/status-im/status-go/services/accounts
        imports github.com/status-im/status-go/protocol
        imports github.com/status-im/status-go/protocol/sqlite
        imports github.com/status-im/status-go/protocol/migrations: no Go files in /private/tmp/nix-build-status-go-develop-230e05c-android.drv-0/go/src/github.com/status-im/status-go/protocol/migrations
package gobind
        imports github.com/status-im/status-go/mobile
        imports github.com/status-im/status-go/api
        imports github.com/status-im/status-go/node
        imports github.com/status-im/status-go/services/accounts
        imports github.com/status-im/status-go/protocol
        imports github.com/status-im/status-go/protocol/sqlite
        imports github.com/status-im/status-go/protocol/pushnotificationclient/migrations: no Go files in /private/tmp/nix-build-status-go-develop-230e05c-android.drv-0/go/src/github.com/status-im/status-go/protocol/pushnotificationclient/migrations
package gobind
        imports github.com/status-im/status-go/mobile
        imports github.com/status-im/status-go/api
        imports github.com/status-im/status-go/node
        imports github.com/status-im/status-go/services/accounts
        imports github.com/status-im/status-go/protocol
        imports github.com/status-im/status-go/protocol/sqlite
        imports github.com/status-im/status-go/protocol/pushnotificationserver/migrations: no Go files in /private/tmp/nix-build-status-go-develop-230e05c-android.drv-0/go/src/github.com/status-im/status-go/protocol/pushnotificationserver/migrations
package gobind
        imports github.com/status-im/status-go/mobile
        imports github.com/status-im/status-go/api
        imports github.com/status-im/status-go/node
        imports github.com/status-im/status-go/services/accounts
        imports github.com/status-im/status-go/protocol
        imports github.com/status-im/status-go/protocol/sqlite
        imports github.com/status-im/status-go/protocol/transport/migrations: no Go files in /private/tmp/nix-build-status-go-develop-230e05c-android.drv-0/go/src/github.com/status-im/status-go/protocol/transport/migrations
../../go/src/github.com/status-im/status-go/services/wallet/router/pathprocessor/processor_bridge_celar.go:27:2: cannot find package "github.com/status-im/status-go/services/wallet/router/pathprocessor/cbridge" in any of:
        /private/tmp/nix-build-status-go-develop-230e05c-android.drv-0/go/src/github.com/status-im/status-go/vendor/github.com/status-im/status-go/services/wallet/router/pathprocessor/cbridge (vendor tree)
        /nix/store/3l7m8vni45h2inv7fcifvp6izxm243an-go-1.21.10/share/go/src/github.com/status-im/status-go/services/wallet/router/pathprocessor/cbridge (from $GOROOT)
        /private/tmp/nix-build-status-go-develop-230e05c-android.drv-0/gomobile-work/src/github.com/status-im/status-go/services/wallet/router/pathprocessor/cbridge (from $GOPATH)
        /private/tmp/nix-build-status-go-develop-230e05c-android.drv-0/go/src/github.com/status-im/status-go/services/wallet/router/pathprocessor/cbridge
        /nix/store/rzyfh6ibdc8q0s13g0z0sb2sxbnjzlh1-gomobile-unstable-2022-05-18/src/github.com/status-im/status-go/services/wallet/router/pathprocessor/cbridge
../../go/src/github.com/status-im/status-go/services/connector/test_helpers.go:15:2: cannot find package "github.com/status-im/status-go/rpc/mock/client" in any of:
        /private/tmp/nix-build-status-go-develop-230e05c-android.drv-0/go/src/github.com/status-im/status-go/vendor/github.com/status-im/status-go/rpc/mock/client (vendor tree)
        /nix/store/3l7m8vni45h2inv7fcifvp6izxm243an-go-1.21.10/share/go/src/github.com/status-im/status-go/rpc/mock/client (from $GOROOT)
        /private/tmp/nix-build-status-go-develop-230e05c-android.drv-0/gomobile-work/src/github.com/status-im/status-go/rpc/mock/client (from $GOPATH)
        /private/tmp/nix-build-status-go-develop-230e05c-android.drv-0/go/src/github.com/status-im/status-go/rpc/mock/client
        /nix/store/rzyfh6ibdc8q0s13g0z0sb2sxbnjzlh1-gomobile-unstable-2022-05-18/src/github.com/status-im/status-go/rpc/mock/client

error: builder for '/nix/store/qzi5y7i1p2qqd7g0g0xz767fcxaxsd9y-status-go-develop-230e05c-android.drv' failed with exit code 1;
       last 10 log lines:
       >      /private/tmp/nix-build-status-go-develop-230e05c-android.drv-0/gomobile-work/src/github.com/status-im/status-go/services/wallet/router/pathprocessor/cbridge (from $GOPATH)
       >    /private/tmp/nix-build-status-go-develop-230e05c-android.drv-0/go/src/github.com/status-im/status-go/services/wallet/router/pathprocessor/cbridge
       >      /nix/store/rzyfh6ibdc8q0s13g0z0sb2sxbnjzlh1-gomobile-unstable-2022-05-18/src/github.com/status-im/status-go/services/wallet/router/pathprocessor/cbridge
       > ../../go/src/github.com/status-im/status-go/services/connector/test_helpers.go:15:2: cannot find package "github.com/status-im/status-go/rpc/mock/client" in any of:
       >         /private/tmp/nix-build-status-go-develop-230e05c-android.drv-0/go/src/github.com/status-im/status-go/vendor/github.com/status-im/status-go/rpc/mock/client (vendor tree)
       >       /nix/store/3l7m8vni45h2inv7fcifvp6izxm243an-go-1.21.10/share/go/src/github.com/status-im/status-go/rpc/mock/client (from $GOROOT)
       >      /private/tmp/nix-build-status-go-develop-230e05c-android.drv-0/gomobile-work/src/github.com/status-im/status-go/rpc/mock/client (from $GOPATH)
       >         /private/tmp/nix-build-status-go-develop-230e05c-android.drv-0/go/src/github.com/status-im/status-go/rpc/mock/client
       >   /nix/store/rzyfh6ibdc8q0s13g0z0sb2sxbnjzlh1-gomobile-unstable-2022-05-18/src/github.com/status-im/status-go/rpc/mock/client
       >
       For full logs, run 'nix log /nix/store/qzi5y7i1p2qqd7g0g0xz767fcxaxsd9y-status-go-develop-230e05c-android.drv'.
make: *** [status-go-android] Error 1
```

A solution to this problem would be to build the generated files in a `preBuild` Phase

--- a/nix/status-go/library/default.nix
+++ b/nix/status-go/library/default.nix
@@ -1,4 +1,5 @@
-{ lib, stdenv, meta, source, buildGoPackage }:
+{ lib, stdenv, meta, source, buildGoPackage,
+  go-bindata, mockgen, protoc-gen-go, protobuf3_20 }:
 
 buildGoPackage {
   pname = source.repo;
@@ -6,6 +7,10 @@ buildGoPackage {
 
   inherit meta;
   inherit (source) src goPackagePath;
+
+  nativeBuildInputs = [
+    go-bindata mockgen protoc-gen-go protobuf3_20
+  ];
 
   phases = ["unpackPhase" "configurePhase" "buildPhase"];
 
@@ -22,6 +27,7 @@ buildGoPackage {
   preBuild = ''
     pushd go/src/$goPackagePath
     go run cmd/library/*.go > $NIX_BUILD_TOP/main.go
+    make generate SHELL=$SHELL GO111MODULE=on GO_GENERATE_CMD='go generate'
     popd
   '';
 

--- a/nix/status-go/mobile/build.nix
+++ b/nix/status-go/mobile/build.nix
@@ -1,5 +1,6 @@
 { callPackage, lib, buildGoPackage
 , androidPkgs, openjdk, gomobile, xcodeWrapper, removeReferencesTo
+, go-bindata, mockgen, protobuf3_20, protoc-gen-go
 , meta
 , source
 , platform ? "android"
@@ -15,7 +16,7 @@ let
   isAndroid = platform == "android";
   enforceXCodeAvailable = callPackage ./enforceXCodeAvailable.nix { };
 
-in buildGoPackage {
+in buildGoPackage rec {
   pname = source.repo;
   version = "${source.cleanVersion}-${source.shortRev}-${platform}";
 
@@ -27,8 +28,9 @@ in buildGoPackage {
   __noChroot = isIOS;
 
   extraSrcPaths = [ gomobile ];
-  nativeBuildInputs = [ gomobile removeReferencesTo ]
-    ++ optional isAndroid openjdk
+  nativeBuildInputs = [
+    gomobile removeReferencesTo go-bindata mockgen protoc-gen-go protobuf3_20
+  ] ++ optional isAndroid openjdk
     ++ optional isIOS xcodeWrapper;
 
   ldflags = goBuildLdFlags;
@@ -43,6 +45,13 @@ in buildGoPackage {
   # https://github.com/status-im/status-mobile/issues/19581
   # TODO: try removing when go is upgraded to 1.22
   GODEBUG = "netdns=cgo+2";
+
+  preBuild = ''
+    echo 'Generate static files'
+    pushd go/src/$goPackagePath
+    make generate SHELL=$SHELL GO111MODULE=on GO_GENERATE_CMD='go generate'
+    popd
+  '';
 
   buildPhase = ''
     runHook preBuild

--- a/status-go-version.json
+++ b/status-go-version.json
@@ -3,7 +3,7 @@
     "_comment": "Instead use: scripts/update-status-go.sh <rev>",
     "owner": "status-im",
     "repo": "status-go",
-    "version": "v2.3.0",
-    "commit-sha1": "effde33d493ff52b66c8013067c9aa6444d84473",
-    "src-sha256": "1rd0mczdkvjm9pbmza234v1jy1bvqh3ni2xck1l42ha4sld2aqjl"
+    "version": "develop",
+    "commit-sha1": "ba37c32c07d1b05a94d1c7f7cb1b38fb82730f7b",
+    "src-sha256": "1f1bbw70iircicda9w0hch56hxr7p1vvzkrk58v8p5sq23xzdf6y"
 }

